subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/consistency/ProjectLocalDependencyResolutionConsistencyIntegrationTest.groovy

```diff
@@ -0,0 +1,300 @@
+                edge('org:foo:{prefer 1.1}', 'org:foo:1.0') {
+                    byConstraint("version resolved in configuration ':other' by consistent resolution")
+                }
+                constraint("org:foo:{strictly 1.0}", "org:foo:1.0")
+            }
+        }
+    }
+
+    def "a transitive dependency may be downgraded by consistent resolution"() {
+        repository {
+            'org:foo:1.0' {
+                dependsOn 'org:fooA:1.0'
+            }
+            'org:bar:1.0' {
+                dependsOn 'org:barA:1.0'
+            }
+            'org:fooA:1.0' {
+                dependsOn 'org:transitive:1.0'
+            }
+            'org:barA:1.0' {
+                dependsOn 'org:transitive:2.0'
+            }
+            'org:transitive:1.0'()
+            'org:transitive:2.0'()
+        }
+
+        buildFile << """
+            configurations {
+                implementation
+                runtimeOnly.extendsFrom(implementation)
+                compileClasspath.extendsFrom(implementation)
+                runtimeClasspath.extendsFrom(implementation, runtimeOnly)
+                runtimeClasspath {
+                   resolveConsistentlyWith(compileClasspath)
+                }
+            }
+
+            dependencies {
+                implementation 'org:foo:1.0'
+                runtimeOnly 'org:bar:1.0'
+            }
+        """
+        def resolve = new ResolveTestFixture(buildFile, "runtimeClasspath")
+        resolve.expectDefaultConfiguration("runtime")
+        resolve.prepare()
+        when:
+        repositoryInteractions {
+            'org:foo:1.0' {
+                expectResolve()
+            }
+            'org:bar:1.0' {
+                expectResolve()
+            }
+            'org:fooA:1.0' {
+                expectResolve()
+            }
+            'org:barA:1.0' {
+                expectResolve()
+            }
+            'org:transitive:1.0' {
+                expectResolve()
+            }
+        }
+        succeeds 'checkDeps'
+
+        then:
+        resolve.expectGraph {
+            root(':', ':test:') {
+                module('org:foo:1.0') {
+                    byConstraint("version resolved in configuration ':compileClasspath' by consistent resolution")
+                    module('org:fooA:1.0') {
+                        byConstraint("version resolved in configuration ':compileClasspath' by consistent resolution")
+                        module("org:transitive:1.0") {
+                            byConstraint("version resolved in configuration ':compileClasspath' by consistent resolution")
+                        }
+                    }
+                }
+                module('org:bar:1.0') {
+                    module('org:barA:1.0') {
+                        edge("org:transitive:2.0", "org:transitive:1.0") {
+                            byConstraint("version resolved in configuration ':compileClasspath' by consistent resolution")
```

Comment:
would it make sense to extract those out similar to `selectedByRule` on `ResolveTestFixture`

Reply:
yup, done!